internal/controller/accessmanagement_controller.go

```diff
@@ -252,30 +295,54 @@ func (r *TemplateManagementReconciler) createTemplateChain(ctx context.Context,
 	return nil
 }
 
-func (r *TemplateManagementReconciler) deleteTemplateChain(ctx context.Context, chain templateChain) error {
+func (r *AccessManagementReconciler) createCredential(ctx context.Context, namespace, name string, spec *hmc.CredentialSpec) error {
+	l := ctrl.LoggerFrom(ctx)
+
+	target := &hmc.Credential{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				hmc.HMCManagedLabelKey: hmc.HMCManagedLabelValue,
+			},
+		},
+		Spec: *spec,
+	}
+	err := r.Create(ctx, target)
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			return nil
+		}
+		return err
+	}
```

Comment:
nit: could be simplified to (and applicable above/below depending on the `Ignore*` function), up to you whether to apply/skip

```go

	if err := r.Create(ctx, target); client.IgnoreAlreadyExists(err) != nil {
		return err
	}
```